### PR TITLE
apis: add Prod overcommit strategy api

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

Add the API of the Prod resource overcommitment strategy.

### Ⅱ. Does this pull request fix one issue?

Part of #1428.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/extension/slo_controller_config.go [hunk 1]
```diff
@@ -214,6 +233,28 @@ type ColocationStrategy struct {
 	// MidMemoryAllocatable <= NodeMemoryAllocatable * MidMemoryThresholdPercent / 100.
 	MidMemoryThresholdPercent *int64 `json:"midMemoryThresholdPercent,omitempty" validate:"omitempty,min=0,max=100"`
 
+	// ProdOvercommitPolicy specifies the policy for Prod overcommitment.
```
Comment (zwzhang0107) [inline apis/extension/slo_controller_config.go:236]: how about merge into one struct

Comment (saintube) [reply apis/extension/slo_controller_config.go]: done

**Pull request closed without merging**